### ISSUE TITLE
Error when parsing messages

### ISSUE THREAD

**Issue by mtushe1** [ownaginatious/fbchat-archive-parser#57]
```
Traceback (most recent call last):                                               
  File "/Users/miroslav/anaconda3/bin/fbcap", line 11, in <module>
    sys.exit(fbcap())
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/fbchat_archive_parser/main.py", line 127, in fbcap
    write(fmt, fbch, directory or sys.stdout)
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/fbchat_archive_parser/writers/__init__.py", line 44, in write
    selected_writer().write(data, stream_or_dir)
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/fbchat_archive_parser/writers/writer.py", line 17, in write
    return self.write_history(data, stream)
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/fbchat_archive_parser/writers/text.py", 
line 19, in write_history
    ('-' * len(history.user)) + "-\n"
TypeError: object of type 'NoneType' has no len()
```

Happens every time regardless. Does anyone have any idea what's wrong?

**Comment by ownaginatious** [comment]
So it looks like the error is that your `messages.htm` file was somehow missing the user (a.k.a. your name). If you open the file, you should look something like this:

```html
    ...
    <div class="contents">
        <h1>Your Name</h1>
        <p><a href="../messages/....html">Friend A</a></p>
        <p><a href="../messages/....html">Friend B</a></p>
        <p><a href="../messages/....html">Friend C</a></p>
        <p><a href="../messages/....html">Friend D</a></p>
        <p><a href="../messages/....html">Friend E</a></p>
    ...
```

Do you see something different?

**Comment by ownaginatious** [comment]
Oh, I think I know the error. It seems you tried to parse one of the HTML files under `messages/`. You should be parsing the `messages.htm` file.

```
├── html/
│   ├── ...
│   ├── messages.htm <- Parse me!
├── messages/
│   ├── 1043...html
│   ├── ...
```

**Comment by mtushe1** [comment]
Oops, my bad

**Comment by mtushe1** [comment]
Ok, now I'm getting something different

```
Unable to locate the referenced chat file "/Users/miroslav/Downloads/facebook/html/sages/338103939874843.html". Please ensure that your "messages.htm" file is relative to your "messages/" directory in the following way while parsing:

    ├── html/
    │   ├── ...
    │   ├── messages.htm
    ├── messages/

```

Although if I open `messages.htm `via a browser and try to navigate to my messages - it works, no broken links.

**Comment by ownaginatious** [comment]
Hmm, I think I may have made some assumptions about how the `messages.htm` file is formatted that aren't always true. What do the `<a>` tags look like in your `messages.htm` file? Do they look like this?

```html
<p><a href="../messages/<some long number>.html">Friend A</a></p>
```
Or like this?

```html
<p><a href="messages/<some long number>.html">Friend A</a></p>
```

**Comment by mtushe1** [comment]
@ownaginatious 

`</h1><p><a href="messages/338103939874843.html">Friend A </a></p><p>`

**Comment by ownaginatious** [comment]
Hmm, odd. I just downloaded my data again and it looks like Facebook broke the relative path...

Anyway, I fixed it to be compatible with that format and the old format (should they ever try to fix it...). Please try the latest version: `1.1.post6`

**Comment by mtushe1** [comment]
sorry, still the same error...

**Comment by ownaginatious** [comment]
Did you do `pip install --upgrade fbchat-archive-parser`?

**Comment by mtushe1** [comment]
Yes

**Comment by ownaginatious** [comment]
Can you post a stack trace?

**Comment by ownaginatious** [comment]
Yeah, it works for me. If you could post the error output, I could at least see which directory it's trying to look in :/

**Comment by mtushe1** [comment]
```
Traceback (most recent call last):
  File "/Users/miroslav/anaconda3/lib/python3.6/site-packages/fbchat_archive_parser/parser.py", line 433, in process_thread
    with io.open(file_path, 'rt', encoding='utf8') as thread_file:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/miroslav/Downloads/facebook-miroslavtushev/html/messages/338103939874843.html'

Unable to locate the referenced chat file "/Users/miroslav/Downloads/facebook-miroslavtushev/html/messages/338103939874843.html". Please ensure that your "messages.htm" file is relative to your "messages/" directory in the following way while parsing:

    ├── html/
    │   ├── ...
    │   ├── messages.htm
    ├── messages/

```

**Comment by ownaginatious** [comment]
Well, only thing I can think of at this point is that the implementation of some of the Python `os` library functions work differently on your computer that mine. I tried switching to something that should be more reliable. Hopefully it'll work now: `1.1.post7`

**Comment by mtushe1** [comment]
Works like a charm now, many many thanks to you for guiding me thru this process!!!

**Comment by ownaginatious** [comment]
Great, I'm glad that fixed it! :D

**Comment by eldenvo** [comment]
Hi there, I'm definitely using 1.1.post7 and I'm also getting the same thing:

`
Unable to locate the referenced chat file "messages/997041420387057.html". Please ensure that your "messages.htm" file is relative to your "messages/" directory in the following way while parsing:

    ├── html/
    │   ├── ...
    │   ├── messages.htm
    ├── messages/
`

**Comment by ownaginatious** [comment]
Which OS are you using?

**Comment by eldenvo** [comment]
Windows 10

**Comment by ownaginatious** [comment]
Hmm, it’s probably an issue with how the tool uses Python file system tools and Windows paths. This tool is unfortunately not well tested with Windows. I will look into it soon (gotta get a Windows VM running first).

**Comment by eldenvo** [comment]
No problem and thanks for your work - it has generally worked fine for me up until this change in structure!

**Comment by ownaginatious** [comment]
^ should be fixed now in `1.1.post8` :+1:
